Use dch instead of hack to generate changelog

ann0see opened the pull request:
See: https://github.com/nefarius2001/jamulus/blob/autobuild/stable2/distributions/build-debian-package-auto.sh

Probably a lot more could be improved here.

pljones commented on distributions/build-debian-package-auto.sh:
```diff
@@ -2,30 +2,18 @@
 cd ..
 
 # get the jamulus version from pro file
 VERSION=$(cat Jamulus.pro | grep -oP 'VERSION = \K\w[^\s\\]*')
```
We should have an autobuild script to do this in a consistent manner...

pljones commented:
Is there an example of what `dch` produces you can paste?

ann0see commented:
dch produces a debian changelog file. It's not really useable:

```

jamulus (3.8.0rc1dev) UNRELEASED; urgency=medium

  [ marc ]
  * Initial release.

  [ ann0see ]
  *   - Mac: Updated to use Qt 5.12.10 for better compatibility with Big
    Sur (\#1687).   This drops support for Yosemite and El Capitan in the
    distributed version.   (contributed by \@softins)  - Code: Added
    automatic code formatting with clang-format (\#901, \#1127).
    (contributed by \@passing)  - GUI: Added "About Qt" to the help menu
    to display version of Qt (\#1685, \#1692).   (contributed by \@softins)
    - GUI: Corrected the operation of What's This in the settings dialog
    (\#1622, \#1635).   (contributed by \@dcorson-ticino-com)  - GUI:
    Improved the muted speaker icon display (#1691).   (contributed by
    \@henkdegroot)  - Added new icons for Linux desktop use (\#1672).
    (contributed by \@jujudusud)  - Server: Improvements to multi-
    threading performance (\#960)   (contributed by \@menzels, \@softins)  -
    Github autobuild for Mac now uses Xcode 11.7 and SDK 10.15 for
    compatibility   with Qt5 (\#1655).   (contributed by \@softins)  -
    GUI: Alignment of delay measurements display improved (\#1606,
    \#1617).   (contributed by \@pljones)  - GUI: Corrected handling of
    custom directory server in the server, to prevent   unintended
    registration with a directory server (\#1624, \#1627).   (contributed
    by \@softins)  - GUI: Corrected alignment of Mute icon above fader
    (\#811, \#1312, \#1640).   (contributed by \@vimpostor)  - The term
    "Central server" has been replaced with "Directory server" (\#1407).
    Note that the program will still accept the --centralserver option
    for backward   compatibility with existing system startup scripts.
    (contributed by \@pljones)  - GUI: Settings window has been
    reorganized into tabs (\#1415, \#1554, \#1542):     * User Profile
    window has been integrated into the settings window.     * Input Pan
    has been moved to the newly created Advanced tab and removed from
    main window.     * Network and stats have been moved to the right in
    the Settings window.   (contributed by \@dcorson-ticino-com,
    \@pljones)  - GUI: Support for more than two mixer rows has been
    added (\#1549, \#1560).   (contributed by \@pljones)  - GUI: --
    clientname also works in dialog titles now (\#1352, 1370).
    (contributed by \@dcorson-ticino-com)  - GUI: Translations have been
    updated   * Dutch, by \@henkdegroot (\#1562, \#1623)   * French, by
    \@jujudusud (\#1648)   * German, by \@rolamos (\#1677)   * Italian, by
    \@dzpex (\#1620)   * Polish, by \@SeeLook (\#1619)   * Portuguese
    Brazilian, by \@melcon (\#1671)   * Portuguese European, by \@Snayler
    (\#1689)   * Slovak, by \@jose1711 (\#1647)   * Spanish, by \@ignotus666
    (\#1621)   * Swedish, by \@genesisproject2020 (\#1664, \#1696)  -
    Client: Automatic channel fader adjustment simplifies mixer setup by
    using the channel level meters (\#1071).   (contributed by
    \@JohannesBrx)  - Client: Basic audio feedback detection has been
    added (\#1179).   (contributed by \@JohannesBrx)  - Network: Support
    for DSCP Quality of Service flags has been added (\#1310).   This is
    supposed to lead to improved network performance.   It is enabled by
    default.   On Windows, this requires additional configuration in
    order to work.   (contributed by \@DavidSavinkoff)  - Client: Support
    for input gain boost has been added (\#1222, \#1030)   (contributed by
    \@hoffie)  - Client: Grouping support has been extended to allow for
    up to eight groups (\#1551).   (contributed by \@pljones)  - Client: A
    hint regarding non-default Jack support has been added (\#1397,
    \#1438).   (contributed by \@djfun)  - Server: Support for Delay
    Panning has been added (\#332, \#567, \#1151, \#1417):   This feature
    can be enabled on servers using the new --delaypan option.
    Enabling this feature will slightly increase server CPU usage.   It
    can create a much more realistic spatial sound impression for
    clients   that are set to Stereo or Mono-in/Stereo-out mode.
    (contributed by \@DetlefHennings, \@Hk1020, \@softins)  - Server: Half-
    connected clients will no longer receive audio (\#1243, \#1589):
    Note: This breaks compatibility with client versions before 3.3.0
    (Feb 2013).   If you update your server, ensure that all clients use
    3.3.0 or later as well.   (contributed by \@softins)  - Server: HTML
    status file is now emptied on exit (\#1423, \#1427)   (contributed by
    \@hoffie, \@drummer1154)  - Server: An explicit bind address can now
    be specified (\#141, \#1561).   This can be done by the new --
    serverbindip option.   (contributed by \@buv)  - Recorder:
    Compatibility with third-party tools such as Audacity has been
    improved (\#1384, \#1424, \#1437).   Non-ASCII characters are now
    stripped out when creating filenames.   (contributed by \@softins,
    \@gilgongo, \@reinhardwh)  - Recorder: Failures to start recording no
    longer result in crashes (\#1163, \#1289, \#1463)   (contributed by
    \@hoffie, \@softins, \@pljones)  - Recorder: Logging has been improved
    (\#1284, \#1463).   (contributed by \@hoffie, \@drummer1154)  - Bug fix:
    The mute indicator is now also shown in mono mode (\#1074).
    (contributed by \@npostavs)  - Bug fix: Channel mapping on Windows
    now only resets if the sound card or the numbers of channels changed
    (\#1347).   (contributed by \@ann0see)  - Bug fix: Jamulus client
    should now close itself correctly if a non-working device was
    selected (\#872).   (contributed by \@npostavs)  - Bug fix: Server no
    longer crashes while changing the recording directory (\#1501,
    \#1573).   (contributed by \@henkdegroot, \@hoffie, \@softins)  - Bug
    fix: Example systemd unit has been fixed to avoid crashing when
    sending signals (\#1515, \#1518).   (contributed by \@softins,
    \@helgeerbe, \@gilgongo)  - Bug fix: The Windows installer now
    correctly compiles in a path with spaces (\#864, \#1319).
    (contributed by \@henkdegroot)  - Performance: Opus encoding/decoding
    now uses machine-specific optimizations (\#1105).   (contributed by
    \@npostavs)  - Performance: Timer configuration for Windows servers
    has been improved (\#1536).   (contributed by \@npostavs)  - iOS
    support is being worked on (\#1450)   (contributed by \@jeroenvv)  -
    Build: Creation of debug builds has been simplified (\#1516).
    (contributed by \@hoffie)  - Internal constants for Jack usage have
    been renamed (\#1429).   (contributed by \@djfun)  - Internal legacy
    IP address variables have been cleaned up (\#1400).   (contributed by
    \@wferi)

 -- ann0see <ann0see@local.host>  Thu, 27 May 2021 19:27:38 +0200
```

hoffie commented:
I don't know offhand what the current state looks like, but the new one does not look like what I had expected either (properly indented lists)?

ann0see commented:
Yeah. But it's better than a syntactically wrong changelog file.

ann0see commented:
We could loop through each change log entry but this would mean some rewriting of this file: https://github.com/jamulussoftware/jamulus/blob/master/.github/actions_scripts/getChangelog.pl

Probably it can be done with some regex magic which I'm not quite into.

hoffie reviewed (approved):
I'm approving on the basis that it improves things, not on the basis that I'm happy with the result. ;)

Maybe defer merging for some days. I can try to have a look at the supposed regexp magic. :)

ann0see commented:
@pljones do we want this to be merged for the next release or not?

pljones commented:
I'd like to see any proposed final result -- any automated process will only be as good as the human input to that process...

npostavs commented:
> Yeah. But it's better than a syntactically wrong changelog file.

Is it though? I'm not sure how a non-readable changelog file can be better.

Anyway, I tried the loop over each change method, and it seems to work okay. But I'm not sure what the deal with the "Initial Release" entry is. Code is here: https://github.com/jamulussoftware/jamulus/compare/master...npostavs:np.dch-changelog, the output looks like this: [changelog.dch.log](https://github.com/jamulussoftware/jamulus/files/7598066/changelog.dch.log)

ann0see commented:
>  But I'm not sure what the deal with the "Initial Release" entry is

Probably the changelog would in theory include all changelogs of previous versions. I assume @mirabilos our debian packager knows more about how to have this file syntactically correct. @npostavs feel free to open a PR

npostavs commented:
I opened #2138.

ann0see commented:
Ok. Closing this then! Thanks
